elfloader-tool/src/arch-riscv/boot.c

```diff
@@ -89,11 +91,6 @@ static int map_kernel_window(struct image_info *kernel_info)
 
     /* Map the elfloader into the new address space */
 
-    if (!IS_ALIGNED((uintptr_t)_text, PT_LEVEL_2_BITS)) {
```

Comment:
could you explain why the check existed previously and why it can be removed a bit?

Reply:
Because the kernel also could only handle setting up the kernel mappings if the image was on a page boundary and because the elfloader was always placed on a page boundary by the bbl/opensbi implementation. The rest of map_kernel_window was specialized to these assumptions and so had the checks to catch when there was divergence. Now that divergence can be handled by the code, the checks should be able to be removed.